statsd/src/main/java/com/avast/metrics/statsd/StatsDCounter.java

```diff
@@ -0,0 +1,54 @@
+package com.avast.metrics.statsd;
+
+import com.avast.metrics.api.Counter;
+import com.timgroup.statsd.StatsDClient;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+public class StatsDCounter implements Counter {
+    private final StatsDClient client;
+    private final String name;
+
+    private final AtomicLong count = new AtomicLong(0);
+
+    StatsDCounter(final StatsDClient client, final String name) {
+        this.client = client;
+        this.name = name;
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public void inc() {
+        count.incrementAndGet();
+        client.count(name, 1);
+    }
+
+    @Override
+    public void inc(final long n) {
+        count.addAndGet(n);
+        client.count(name, n);
+    }
+
+    @Override
+    public void dec() {
+        count.decrementAndGet();
+        client.count(name, -1);
+    }
+
+    @Override
+    public void dec(final int n) {
+        final int delta = -n;
+
+        count.addAndGet(delta);
+        client.count(name, delta);
+    }
+
+    @Override
+    public long count() {
```

Comment:
I understood. We actually do the same thing but just using JMX for that.

Try to look at it this way. The interface `Counter` represents counter of some events and whoever uses it can ask it for the current value. The user of that interface does not know that it's actually going to StatsD and that it's actually running in distributed environment and the counter is actually aggregated somewhere. That's completely different issue. So when you ask for the current value I would argue you expect to get only the local value, nothing more. Let's say you would count the number of concurrent requests (per node) - naturally it would work for this use case. I really don't see a problem here.